packages/common/src/interfaces.ts

```diff
@@ -40,8 +40,9 @@ export type AccessList = AccessListItem[]
 
 export interface StateManagerInterface {
   accountExists(address: Address): Promise<boolean>
+  accountIsEmptyOrNonExistent(address: Address): Promise<boolean>
```

Comment:
Short context addition: I added (+ named) this method in a first round, @jochem-brouwer feel very free to rename or remove.

So this is the "account being dead" definition from [EIP-161](https://eips.ethereum.org/EIPS/eip-161), that's the state trie clearing HF included in Spurious Dragon. One alternative was also to name as `accountisDead()` or take the EIP in, `accountIsDeadEIP161()` or so, didn't like both versions so I chose this more declarative one.

That said: I would also be very much ok if we totally remove and sleeken the interface here and do the explicit check Gabriel suggested, would have a slight preference for this but totally no strong opinion.

So very much up to you! 🙂

Reply:
I will check in what contexts this is used (and I doubt there are a lot). We might just remove this and then use it, at the context, by either checking `getAccount(addr) === undefined || getAccount(addr).IsEmpty()` (of course written down a bit more stylish). I think the only context relevant is the state clearing, and I do not think that there is an external use case. There would be if you would check if the account exists, but since all dead accounts are cleared (no dead accounts exists on mainnet) this also doesn't seem to have any external use case.